Atlas.DonorImport.Test/Services/DonorChecker/DonorInfoCheckerTests.cs

```diff
@@ -170,5 +170,18 @@ public async Task CompareDonorInfoInFileToAtlasDonorStore_WhenEmptyDonorFileExce
             await notificationSender.Received().SendAlert("Donor info checker file was present but it was empty.", $"Donors file: name-of-the-file.ext", Priority.Medium, Arg.Any<string>());
         }
 
+        [Test]
+        public async Task CompareDonorInfoInFileToAtlasDonorStore_DisposeIsCalledOnILazilyParsingDonorFile()
+        {
+            var file = DonorImportFileBuilder.NewWithoutContents.With(x => x.FileLocation, "name-of-the-file.ext");
+            var fileParserResult = Substitute.For<ILazilyParsingDonorFile>();
+            fileParserResult.ReadLazyDonorUpdates().Throws<EmptyDonorFileException>();
+            fileParser.PrepareToLazilyParseDonorUpdates(default).Returns(fileParserResult);
```

Comment:
This set up needs explaining - why are you intentionally throwing an exception on read of the file?

I suspect this is copy/paste error.